src/api/cartApi.ts

```diff
@@ -0,0 +1,82 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { projectKey, apiUrl } from './constants';
+import type { Cart, LineItemDraft } from '../types/cartApi';
+import type { Response } from '../types/productsApi';
+
+function isCartListResponse(data: unknown): data is Response<Cart> {
+  return (
+    typeof data === 'object' && data !== null && 'results' in data && Array.isArray(data.results)
+  );
+}
+
+export const cartApi = createApi({
+  reducerPath: 'cartApi',
+  baseQuery: fetchBaseQuery({
+    baseUrl: `${apiUrl}/${projectKey}`,
+    prepareHeaders: async (headers) => {
+      //todo: handle token of anonymous or customer user when this will implemented
+      const accessToken = localStorage.getItem('auth_token');
+      headers.set('Authorization', `Bearer ${accessToken}`);
+      return headers;
+    },
+  }),
+  tagTypes: ['Cart'],
+  endpoints: (build) => ({
+    getMyCarts: build.query<Cart[], void>({
+      query: () => 'me/carts',
+      async transformResponse(response: Response<Cart>) {
+        return response.results;
+      },
+      providesTags: ['Cart'],
+    }),
+
+    getMyActiveCart: build.query({
+      async queryFn(_arguments, _api, _extraOptions, fetchWithBQ) {
```

Comment:
Problem: This queryFn makes several different queries at once, and also contains resource creation logic inside the "get" query. This will make debugging and reuse difficult.  
Solution: It is better to make a separate createCart endpoint and use it in the component when needed.